src/coprocessor/dag/expr/err.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{error, io, str};
+use std::string::FromUtf8Error;
+use std::str::Utf8Error;
+use tipb::expression::ScalarFuncSig;
+use tipb::select;
+
+use coprocessor::codec::mysql::Res;
+use util;
+use util::codec::{self, Error as CError};
+
+pub const ERR_UNKNOWN: i32 = 1105;
+pub const ERR_TRUNCATED: i32 = 1265;
+pub const ERR_UNKNOWN_TIMEZONE: i32 = 1298;
+pub const ERR_DATA_OUT_OF_RANGE: i32 = 1690;
+pub const ERR_TRUNCATE_WRONG_VALUE: i32 = 1292;
+
+quick_error! {
+    #[derive(Debug)]
+    pub enum Error {
+        Io(err: io::Error) {
+            from()
+            description("io error")
+            display("I/O error: {}", err)
+            cause(err)
+        }
+        Type { has: &'static str, expected: &'static str } {
+            description("type error")
+            display("type error: cannot get {:?} result from {:?} expression", expected, has)
+        }
+        Codec(err: util::codec::Error) {
+            from()
+            description("codec error")
+            display("codec error: {}", err)
+            cause(err)
+        }
+        ColumnOffset(offset: usize) {
+            description("column offset not found")
+            display("illegal column offset: {}", offset)
+        }
+        UnknownSignature(sig: ScalarFuncSig) {
+            description("Unknown signature")
+            display("Unknown signature: {:?}", sig)
+        }
+        Truncated(s:String) {
+            description("Truncated")
+            display("{}",s)
+        }
+        Overflow(msg:String) {
+            description("Overflow")
+            display("{}",msg)
+        }
+        TruncatedWrongVal(msg:String) {
+            description("TruncatedWrongVal")
+            display("{}",msg)
+        }
+        Eval(s: String,code:i32) {
+            description("evaluation failed")
+            display("{}", s)
+        }
+        Other(err: Box<error::Error + Send + Sync>) {
+            from()
+            cause(err.as_ref())
+            description(err.description())
+            display("unknown error {:?}", err)
+        }
+    }
+}
+
+impl Error {
+    pub fn gen_overflow(data: &str, range: String) -> Error {
```

Comment:
I think you can move these two parameters to the definition in `quick_error` so that `Error::Overflow(data, range)` will act as your `Error::gen_overflow(data, range)` directly.

Reply:
address comments

Reply:
gen_overflow can be removed?